src/manageiq_client/api.py

```diff
@@ -577,28 +582,31 @@ def __call__(self, *args, **kwargs):
         if force_method:
             return getattr(self, force_method.upper())(*args, **kwargs)
 
-        resources = []
-        # We got resources to post
-        for res in args:
-            if isinstance(res, Entity):
-                resources.append(res._ref_repr())
+        if self._method == "post":
+            resources = []
+            # We got resources to post
+            for res in args:
+                if isinstance(res, Entity):
+                    resources.append(res._ref_repr())
+                else:
+                    resources.append(res)
+            query_dict = {"action": self._name}
+            if resources:
+                query_dict["resources"] = []
+                for resource in resources:
+                    new_res = dict(resource)
+                    if kwargs:
+                        new_res.update(kwargs)
+                    query_dict["resources"].append(new_res)
             else:
-                resources.append(res)
-        query_dict = {"action": self._name}
-        if resources:
-            query_dict["resources"] = []
-            for resource in resources:
-                new_res = dict(resource)
                 if kwargs:
-                    new_res.update(kwargs)
-                query_dict["resources"].append(new_res)
-        else:
-            if kwargs:
-                query_dict["resource"] = kwargs
-        if self._method == "post":
+                    query_dict["resource"] = kwargs
+
             result = self.api.post(self._href, **query_dict)
-        elif self._method == "delete":
-            result = self.api.delete(self._href, **query_dict)
+        elif self._method == "patch":
+            result = self.api.patch(self._href, *args)
+        elif self._method in ("delete", "put"):
+            result = getattr(self.api, self._method)(self._href, **kwargs)
```

Comment:
DELETE used to use `query_dict` which I think was effectively `{"resource": kwargs}`, now uses kwargs directly.
does this still work?

actually, does DELETE need a args at all?  Isn't it just url `DELETE /api/:collection/:id` ?

also, is DELETE supported on collection, passing multiple `resources` ids?  It looks like old code supported that but new one doesn't.

Reply:
Nice catch again 👍 ATM it seems body of DELETE is ignored completely, you can send whatever you want and the resource is deleted anyway. I'm passing kwargs just because DELETE can have body according to the HTTP specification, but for our API it's not needed ATM.
DELETE is supported only for single resource, not on collection.